### PR TITLE
DCWL-803-upgraded-to-java-11-and-various-clean-up

### PR DESCRIPTION
**Pull request by ed1286**

### CODE REVIEW

File: test/unit/controllers/actionbuilders/AuthActionSpec.scala [hunk 1]
```diff
@@ -71,17 +71,17 @@ class AuthActionSpec extends UnitSpec with MockitoSugar {
     "authorise as CSP when authorised by auth API and both badge identifier and submitter headers exist" in new SetUp {
       authoriseCsp()
 
-      private val actual: AuthorisedRequest[AnyContentAsXml] = await(authAction.refine(request(testFakeRequestWithMaybeBadgeIdAndMaybeSubmitterId())).right.get)
+      private val actual: AuthorisedRequest[AnyContentAsXml] = await(authAction.refine(request(testFakeRequestWithMaybeBadgeIdAndMaybeSubmitterId())).getOrElse(throw new RuntimeException("failed to refine test authAction")))
```
Comment (kevin-co-hmrc) [inline test/unit/controllers/actionbuilders/AuthActionSpec.scala:74]: I'd prefer to use `inside` (https://www.scalatest.org/user_guide/using_inside) to pattern match on a `Right` to separate the assertion logic from the logic of the system under test.

File: test/integration/CustomsMetricsConnectorSpec.scala [hunk 2]
```diff
@@ -40,15 +40,24 @@ class CustomsMetricsConnectorSpec extends IntegrationTestSpec with GuiceOneAppPe
   private implicit val vpr: ValidatedPayloadRequest[AnyContentAsXml] = TestData.TestCspValidatedPayloadRequest
   private implicit val mockExportsLogger: ExportsLogger = mock[ExportsLogger]
 
-  override protected def beforeAll() {
+  /**
+   * On Jenkins the localhost string is different to when run locally.
+   *
+   * @return
+   */
+  def localhostString(): String = {
```
Comment (kevin-co-hmrc) [inline test/integration/CustomsMetricsConnectorSpec.scala:48]: Seems like a code smell that we have to do this to test log messages :(

File: project/plugins.sbt [hunk 3]
```diff
@@ -3,14 +3,14 @@ resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts
...
 addSbtPlugin("org.scalastyle"    %% "scalastyle-sbt-plugin" % "1.0.0")
 addSbtPlugin("uk.gov.hmrc"       %  "sbt-auto-build"        % "3.9.0")
-addSbtPlugin("net.vonbuchholtz"  %  "sbt-dependency-check"  % "3.1.1")
 addSbtPlugin("ch.epfl.scala"     %  "sbt-scalafix"          % "0.9.18-1")
+addSbtPlugin("com.timushev.sbt"  %  "sbt-updates"            % "0.6.3")
 
-
-
+//To resolve a bug with version 2.x.x of the scoverage plugin - https://github.com/sbt/sbt/issues/6997
+libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
```
Comment (kevin-co-hmrc) [inline project/plugins.sbt:16]: Not needed, the one in `build.sbt` should be sufficient.

**Review by kevin-co-hmrc — CHANGES_REQUESTED** [review]

File: project/plugins.sbt (hunk 3 above)
Comment (ed1286) [reply project/plugins.sbt:16]: removing it breaks it

File: test/integration/CustomsMetricsConnectorSpec.scala (hunk 2 above)
Comment (ed1286) [reply test/integration/CustomsMetricsConnectorSpec.scala:48]: yeah is a weird thing def needs refactoring at some point

File: project/plugins.sbt (hunk 3 above)
Comment (kevin-co-hmrc) [reply project/plugins.sbt:16]: OK

File: test/unit/controllers/actionbuilders/AuthActionSpec.scala (hunk 1 above)
Comment (kevin-co-hmrc) [reply test/unit/controllers/actionbuilders/AuthActionSpec.scala:74]: Fair enough - different strokes.

**Review by kevin-co-hmrc — APPROVED** [review]
Good job 👍🏼

**Comment by ed1286** [comment]
test this please

**Pull request merged**